### DISCUSSION TITLE
"Cannot convert a BigInt value to a number" while calling useContractRead

### DISCUSSION THREAD

**Discussion by cxss** [wevm/wagmi#2348] [Q&A] [answered]
Hi I'm currently migrating an Ethers.js project to wagmi. But when I call useContractRead the error is always thrown:

ContractFunctionExecutionError: An unknown error occurred while executing the contract function "totalSupply".

Raw Call Arguments:
  to:    0x*****
  data:  0x18160ddd
 
Contract Call:
  address:   0x******
  function:  totalSupply()

Docs: https://viem.sh/docs/contract/readContract.html
Details: Cannot convert a BigInt value to a number

My code is :
const { data } = useContractRead({
        address: '0x******',
                abi: [
                    {
                      inputs: [],
                      name: "totalSupply",
                      outputs: [{ name: "", type: "uint256" }],
                      stateMutability: "view",
                      type: "function",
                    },
                  ],
              functionName: 'totalSupply',
        watch: true,
        onError: (err) => { console.error(err) }
      });
 
and no matter I place the abi inline or in a .json file the error keeps the same.
My dependencies is 
"viem": "^0.3.21",
 "wagmi": "^1.0.1",

Can anyone give some help? Thanks!

**Comment by jxom** [comment]
Can you provide a minimal reproducible example? It seems to work in mine: https://stackblitz.com/edit/new-wagmi-obmja7?file=src%2FApp.tsx (ignore the type error – idk why StackBlitz is throwing a type error there).

**Reply by cxss** [reply]
When I downgrade to "wagmi": "0.12.12", the error disappears. I don't know why, but thanks first!

**Reply by jxom** [reply, marked as answer]
Potential fix: https://github.com/hirosystems/stacks.js/issues/1096#issuecomment-946350299

**Comment by ashishbhungani97** [comment]
I had same issue, I just **delete lock file and node modules** , reinstall reslove issue.
